test/python/transpiler/test_csp_layout.py

```diff
@@ -288,6 +288,21 @@ def test_seed(self):
 
         self.assertNotEqual(layout_1, layout_2)
 
+    def test_multi_qubit_gates(self):
+        """multi-qubit gates, like Toffoli, need a complete connection among the involved qubits"""
```

Comment:
Can you also add a test that find the solution?

```suggestion
    def test_multi_qubit_gates(self):
        """Toffoli need a complete connection among the involved qubits."""
        ....

    def test_multi_qubit_gates_fail(self):
        """Toffoli need a complete connection among the involved qubits. No solution on incomplete graph."""
```